packages/web/components/Player/Section/PlayerMeToken.tsx

```diff
@@ -0,0 +1,648 @@
+          status: 'error',
+          isClosable: true,
+        });
+        setLoading(false);
+      });
+  };
+
+  const clearAmounts = () => {
+    setAmount('0');
+    setPreviewAmount('0');
+  };
+
+  const changeTransactionType = () => {
+    clearAmounts();
+    if (transactionType === 'mint') {
+      toggleTransactionType('burn');
+    } else {
+      toggleTransactionType('mint');
+    }
+  };
+
+  const handlePreview = async () => {
+    if (address && amount)
+      await preview(
+        metokenAddress,
+        ethers.utils.parseEther(amount),
+        address,
+        transactionType,
+      ).then((res) => {
+        setPreviewAmount(res);
+      });
+    else {
+      setPreviewAmount('0');
+    }
+  };
+
+  const handleSubmit = async () => {
+    setLoading(true);
+    if (!approved) {
+      await approveMeTokenTx();
+      return;
+    }
+    if (transactionType === 'mint' && amount) {
+      await mint(
+        metokenAddress,
+        ethers.utils.parseEther(amount),
+        owner,
+        provider,
+      )
+        .then(() => {
+          toast({
+            title: 'Success',
+            description: `${meTokenData.symbol} minted successfully.`,
+            status: 'success',
+            isClosable: true,
+          });
+          setLoading(false);
+        })
+        .catch(() => {
+          toast({
+            title: 'Error',
+            description: `${meTokenData.symbol} mint failed.`,
+            status: 'error',
+            isClosable: true,
+          });
+          setLoading(false);
+        });
+    } else {
+      await burn(
+        metokenAddress,
+        ethers.utils.parseEther(amount),
+        owner,
+        provider,
+      )
+        .then(() => {
+          toast({
+            title: 'Success',
+            description: `${meTokenData.symbol} burned successfully.`,
+            status: 'success',
+            isClosable: true,
+          });
+          setLoading(false);
+        })
+        .catch(() => {
+          toast({
+            title: 'Error',
+            description: `${meTokenData.symbol} burn failed.`,
+            status: 'error',
+            isClosable: true,
+          });
+          setLoading(false);
+        });
+    }
+  };
+
+  const handleSetAmount = (swapAmount: string) => {
+    if (swapAmount && !/^[0-9.]+$/.test(swapAmount)) return;
+    if (
+      transactionType === 'burn' &&
+      +swapAmount > +humanizeNumber(meTokenData.balance)
+    ) {
+      setAmount(humanizeNumber(meTokenData.balance));
+    } else {
+      setAmount(swapAmount);
+    }
+  };
+
+  const handleSetSpendAmount = (spendAmount: string) => {
+    if (transactionType === 'mint') {
+      changeTransactionType();
+      handleSetAmount(spendAmount);
+    } else {
+      handleSetAmount(spendAmount);
+    }
+  };
+
+  const handleSwitchToMainnet = async () => {
+    await switchChainOnMetaMask('0x1');
+  };
+
+  const roundNumber = (number: string) =>
```

Comment:
either use an existing function in `utils` or `packages/utils` or move these utility functions there

Reply:
Moved to utils/mathhelpers